pkg/virtualization/core/service/virtualmachine.go

```diff
@@ -272,3 +273,26 @@ func (vmms *VirtualSystemManagementService) SetMemoryMB(vm *virtualsystem.Virtua
 	err = vmms.ModifyVirtualSystemResourceEx(mem.WmiInstance, -1)
 	return
 }
+
+// Re-implementation of Disable-VMConsoleSupport Cmdlet
+// Removes the Synthetic Mouse, Synthetic Keyboard, and Synthetic Display devices from a VM
+func (vmms *VirtualSystemManagementService) RemoveHIDDevices(vm *virtualsystem.VirtualMachine) (err error) {
+	// Get the RASD objects representing the devices. If there are multiple RASD for a device type, will remove the first occurence
+	syntheticMouse, err1 := vm.GetResourceAllocationSettingDataBySubType(virtconstant.SyntheticMouseSubtype)
+	if err1 == nil { // Don't error if failing to get the device. Matches behavior of Disable-VMConsoleSupport
+		defer syntheticMouse.Close()
+		vmms.RemoveVirtualSystemResource(syntheticMouse, -1)
```

Comment:
Do we not care about the error here?

Reply:
We don't want to fail if we error on getting the device. If the device exists remove it, else continue

Reply:
Can we please log it, for debugging purpose